js/main.js

```diff
@@ -1 +1,59 @@
+(function () {
+  function debounce(func, wait, immediate) {
+  	var timeout;
+  	return function() {
+  		var context = this, args = arguments;
+  		var later = function() {
+  			timeout = null;
+  			if (!immediate) func.apply(context, args);
+  		};
+  		var callNow = immediate && !timeout;
+  		clearTimeout(timeout);
+  		timeout = setTimeout(later, wait);
+  		if (callNow) func.apply(context, args);
+  	};
+  };
 
+  function getElementByClassName(className) {
+    if (document.getElementsByClassName) {
+      return document.getElementsByClassName(className)[0];
+    }
+
+    if (document.querySelectorAll) {
+      return document.querySelectorAll((' ' + className).replace(/ +/g, '.'))[0];
```

Comment:
true, but `getElementsByClassName` it's not supported by IE8, that's why it's at the top :)